aten/src/ATen/core/dispatch/Dispatcher.cpp

```diff
@@ -4,10 +4,19 @@
 namespace c10 {
 
 namespace detail {
+
 class RegistrationListenerList final {
 public:
-  void addListener(std::unique_ptr<OpRegistrationListener> listener) {
+  RegistrationHandleRAII addListener(std::unique_ptr<OpRegistrationListener> listener) {
+    auto raw_ptr = listener.get();
     listeners_.push_back(std::move(listener));
+    return RegistrationHandleRAII([this, raw_ptr] {
+        const auto& found_it = std::find_if(listeners_.begin(), listeners_.end(), [raw_ptr](auto& it) {
```

Comment:
Instead of calling find() here, which could be slow, you could make `listeners_` a `std::list` and just remember the iterator in the closure. `std::list` keeps iterators valid, so that's not an issue.

Reply:
Will do in follow up PR